src/plugins/enzymes/src/EnzymesPlugin.cpp

```diff
@@ -74,6 +75,7 @@ EnzymesPlugin::EnzymesPlugin()
     }
 
     EnzymesSelectorWidget::setupSettings();
+    EnzymesLoader::setupEnzymes();
```

Comment:
You can't do it here: it affects UGENE startup time.

If it is fast: do it in the dialog or on the first access. 
If it is slow - there is an extra reason not doing it during startup.

If you do not like that it blocks the dialog: show a loading message before the dialog.

Reply:
I didn't get you logic at all and I am not about to change it. As far as our restriction base grown up, its loading became a bit long - it freezes "Restriction Enzyme" dialog if it loads on the first opening. The best idea is to move it to UGENE loading - as, for example, we validate External Tools or load plugins.

Reply:
You can't compare enzymes file loading on startup & external tools directly. 
Here are the reasons:

1. I invested a lot of work reducing external tools validation time from 10+ seconds to 2-3 seconds and have more plans to reduce it even event more (make it sub-second). This work shows the importance of the problem: UGENE startup must be fast. Startup time is a very important metric.

2. Even current 2-3 seconds needed for ET validation is a 1 time job per version: only the initial run is slow. Your patch makes every run slow.

3. I'm glad you started to improve the restriction sites plugin, but the lazy (delayed) loading was not a miss in the original design but an intentional optimization.

Unfortunately I can't approve the patch (which has a real value) because the patch tries to  fix the unrelated and undiscussed part of the code in the incorrect way.

Reply:
OK